libyul/optimiser/Semantics.cpp

```diff
@@ -93,6 +94,33 @@ void MSizeFinder::operator()(FunctionCall const& _functionCall)
 			m_msizeFound = true;
 }
 
+
+map<YulString, SideEffects> SideEffectsPropagator::sideEffects(
+	Dialect const& _dialect,
+	map<YulString, std::set<YulString>> const& _directCallGraph
+)
+{
+	map<YulString, SideEffects> ret;
+	for (auto const& call: _directCallGraph)
+	{
+		YulString funName = call.first;
+		SideEffects sideEffects;
+		BreadthFirstSearch<YulString>{call.second}.run(
```

Comment:
```suggestion
		BreadthFirstSearch<YulString>{call.second, {funName}}.run(
```
I missed that earlier - otherwise we needlessly visit the function itself, if it's called recursively, right? Shouldn't make much of a difference, though.